File: api/responses/file_responses.py
from database.models.file import File


def file_response(file: File):
    return {
        "id": file.id,
        "model": file.model.value,
        "model_id": file.model_id,
        "file_type": file.file_type.value,
        "file_name": file.file_name,
        "file_location": file.file_location,
        "mime_type": file.mime_type,
        "created_at": file.created_at,
        "updated_at": file.updated_at,
    }

File: api/responses/file_responses.go
package responses

import (
	"time"
	"tyre-match-backend/db/models"
)

type FileResponse struct {
	ID uint `json:"id"`

	Model    string `json:"model"`
	ModelId  uint   `json:"model_id"`
	FileType string `json:"file_type"`

	Name     string `json:"name"`
	Location string `json:"location"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func NewFileResponse(file *models.File) *FileResponse {
	return &FileResponse{
		ID:        file.ID,
		Model:     file.Model,
		ModelId:   file.ModelId,
		FileType:  file.FileType,
		Name:      file.Name,
		Location:  file.Location,
		CreatedAt: file.CreatedAt,
		UpdatedAt: file.UpdatedAt,
	}
}

func NewFileResponses(files []*models.File) []FileResponse {
	var res []FileResponse
	for _, file := range files {
		res = append(res, *NewFileResponse(file))
	}
	return res
}
